Dn02/report/report.jmd

```diff
@@ -0,0 +1,258 @@
+```
+
+and is directly related to the standard normal CDF by
+
+```math
+F(x) = \tfrac{1}{2}\big(1 + \mathrm{erf}(x/\sqrt{2})\big).
+```
+
+Expanding $e^{-t^2}$ as a Maclaurin series and integrating term by term yields the series representation
+
+```math
+\mathrm{erf}(x) = \frac{2}{\sqrt{\pi}} \sum_{n=0}^{\infty} \frac{(-1)^n}{n!\,(2n+1)}\, x^{2n+1}.
+```
+
+
+This series is mathematically valid for all $x$, but when truncated to finitely many terms it only converges quickly near the origin, as shown below with 10 terms.
+
+```julia
+using Dn02
+using Plots
+
+# Plot the Taylor CDF approximation against the true one from Distributions.jl
+using Distributions
+Φ(x) = cdf(Normal(0,1), x)
+
+xs = -3.45:0.1:3.45
+approx = [cdf_taylor(x, terms=10) for x in xs]
+truevals = [Φ(x) for x in xs]
+
+plot(xs, truevals, label="Groundtruth", lw=2, color=:blue)
+plot!(xs, approx, label="Taylor Approximation", lw=2, ls=:dash, color=:red)
+xlabel!("x")
+ylabel!("F(x)")
+title!("Taylor Approximation vs Groundtruth")
+```
+
+
+# Gauss–Legendre Quadrature
+For moderate arguments ($2 \leq |x| \leq 5$), we approximate the standard normal CDF with one step
+
+```math
+F(x) = \frac{1}{\sqrt{2\pi}} \int_{-\infty}^x e^{-t^2/2}\,dt
+```
+
+using Gauss–Legendre quadrature. The idea is to transform the integral to a fixed finite interval, then approximate it by a weighted sum of function evaluations:
+
+```math
+\int_{a}^{b} f(t)\,dt \;\approx\; \sum_{i=1}^{n} w_i f(t_i),
+```
+
+where $t_i$ are the quadrature nodes (roots of Legendre polynomials) and $w_i$ are the corresponding weights. The advantage is that this method achieves high accuracy with relatively few evaluations, keeping the computational cost constant.
+
+
+In our case, the weight function is simply $w(t) = 1$, which makes Gauss–Legendre the appropriate choice (other Gaussian quadratures such as Gauss–Hermite or Gauss–Laguerre are used when exponential weight functions appear).
+
+To ensure constant-time evaluation, we use a fixed rule (10-point Gauss–Legendre), with nodes and weights obtained from published tables and hardcoded into the implementation. Reliable tables can be found online, e.g.:  
+[https://pomax.github.io/bezierinfo/legendre-gauss.html](Gaussian Quadrature Weights and Abscissae).
+
+```julia
+using Dn02
+using Plots
+
+# Plot the Taylor CDF approximation against the true one from Distributions.jl
+using Distributions
+Φ(x) = cdf(Normal(0,1), x)
+
+xs = -3.45:0.1:3.45
+approx = [cdf_legendre(x, n=10) for x in xs]
+truevals = [Φ(x) for x in xs]
+
+plot(xs, truevals, label="Groundtruth", lw=2, color=:blue)
+plot!(xs, approx, label="Legendre Approximation", lw=2, ls=:dash, color=:red)
+xlabel!("x")
+ylabel!("F(x)")
+title!("Legendre Approximation vs Groundtruth")
+```
+
+# Asymptotic Expansion
+
+For large arguments ($x > 5$), we can use an asymptotic expansion to approximate the standard normal CDF. The idea is to express the CDF in terms of the complementary error function:
```

Comment:
The asymptotic expansion is not derived. Where did you find it? Please add the derivation of the expansion or a reference to the source where you found it.